### PR TITLE
mnist test with some bugs

### PR DESCRIPTION
**Pull request by hoppiece**
## MNIST dataset trial
- with MNIST dataset, visualization data does not work 😢 
- The algorithm ends too soon. It ends in the first or second epoch.
- There are so many unknowns in the hyper parameters that it's hard to figure out what's wrong.
Figure_1

### CODE REVIEW

**Comment by hoppiece** [comment]
## comment
- End condition `new_N == old_N` is something wrong, because in many cases topology does not change.
- So I ignored the end condition and continued the epoch until 100. Then I got a little better result.
3_separation

- Unfortunately, this result is not very reproducible, and too slow to caluculate.
### How to run
```
$ python3 mnist_trial.py
```
matplotlib, sklearn, numpy is need.

**Comment by hoppiece** [comment]
- I implemented it as described in the paper, but it is difficult to work well with MNIST data.
- I cannot deny my implementation hasn't some bugs, but the algorithm may have something inadequate.
- I'll try to verify the SGD step and its gradient, and various hyper-parameters.

**Comment by hoppiece** [comment]
- I implemented it as described in the paper, but it is difficult to work well with MNIST data.
- I cannot deny my implementation hasn't some bugs, but the algorithm may have something inadequate.
- I'll try to verify the SGD step and its gradient, and various hyper-parameters.
